FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt

```diff
@@ -627,20 +629,37 @@ class EmailUtil {
     /**
      * Generate a message(new, reply or forward). Don't call it in the main thread.
      *
-     * @param info    The given [OutgoingMessageInfo] which contains information about an outgoing
-     * message.
+     * @param outgoingMsgInfo  The given [OutgoingMessageInfo] which contains information about
+     * an outgoing message.
      * @param pubKeys The public keys which will be used to generate an encrypted part.
      * @return The generated raw MIME message.
      */
-    fun genMessage(context: Context, info: OutgoingMessageInfo, pubKeys: List<String>? = null): Message {
+    fun genMessage(context: Context, accountEntity: AccountEntity,
+                   outgoingMsgInfo: OutgoingMessageInfo): Message {
       val session = Session.getInstance(Properties())
-      return when (info.messageType) {
+      val senderEmail = outgoingMsgInfo.from
+      val senderKeyDetails = SecurityUtils.getSenderKeyDetails(context, accountEntity, senderEmail)
+      var pubKeys: List<String>? = null
+      var prvKeys: List<String>? = null
+      var ringProtector: SecretKeyRingProtector? = null
+
+      if (outgoingMsgInfo.encryptionType === MessageEncryptionType.ENCRYPTED) {
+        val recipients = outgoingMsgInfo.getAllRecipients().toMutableList()
+        pubKeys = SecurityUtils.getRecipientsPubKeys(context, recipients)
+        pubKeys.add(senderKeyDetails.publicKey
+            ?: throw NullPointerException("Sender pub key not found"))
+        prvKeys = listOf(senderKeyDetails.privateKey
+            ?: throw NullPointerException("Sender private key not found"))
```

Comment:
I think IllegalStateException is more appopriate

Reply:
Changed